### DISCUSSION TITLE
Is it possible from defined flags to determine this core against the mbed based core

### DISCUSSION THREAD

**Discussion by davetcc** [earlephilhower/arduino-pico#924] [Q&A] [answered]
I can't see a similar discussion, apologies if there is one. The question seems very basic.

Is there a build flag that is specific to this core that I can use to determine whether this core is selected and not the mbed-based one? Looking at the build flags in the CLion CMake file generated by platformIO it doesn't seem like there are many differences. I also took a quick look through the `Arudino.h` and couldn't see much there either. 

Reason: I'm a library writer and maintain a couple of libraries that really need to know what they are running on in order to perform atomic operations and provide a few cross-platform niceties, it is especially important on RPI pico devices because they are multi-core. I'd like to support this core, as I've got requests to do so, but I also have users that already use the mbed-based core, and I really don't want to break anything for existing users.

**Comment by earlephilhower** [comment, marked as answer]
I would suggest using this one as most other ones seem to all be common (as you seem to have found!):
````
#ifdef  ARDUINO_PICO_REVISION
....
#endif
````

**Reply by davetcc** [reply]
Many thanks, that's good for me.
